vault/data_source_aws_auth_backend_role_tag.go

```diff
@@ -0,0 +1,106 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/hashicorp/vault/api"
+)
+
+func awsAuthBackendRoleTagDataSource() *schema.Resource {
+	return &schema.Resource{
+		Read: awsAuthBackendRoleTagDataSourceRead,
+
+		Schema: map[string]*schema.Schema{
+			"backend": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "aws",
+				Description: "AWS auth backend to read tags from.",
+			},
+			"role": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the role.",
+			},
+			"policies": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Policies to be associated with the tag.",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"max_ttl": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The maximum allowed lifetime of tokens issued using this role.",
+			},
+			"instance_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Instance ID for which this tag is intended. The created tag can only be used by the instance with the given ID.",
+			},
+			"allow_instance_migration": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Allows migration of the underlying instance where the client resides.",
+			},
+			"disallow_reauthentication": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Only allow a single token to be granted per instance ID.",
+			},
+			"tag_value": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tag_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func awsAuthBackendRoleTagDataSourceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	backend := d.Get("backend").(string)
+	role := d.Get("role").(string)
+	path := "auth/" + strings.Trim(backend, "/") + "/role/" + strings.Trim(role, "/") + "/tag"
+
+	data := map[string]interface{}{}
+
+	if v, ok := d.GetOk("policies"); ok {
+		data["policies"] = v
+	}
+	if v, ok := d.GetOk("max_ttl"); ok {
+		data["max_ttl"] = v
+	}
+	if v, ok := d.GetOk("instance_id"); ok {
+		data["instance_id"] = v
+	}
+	if v, ok := d.GetOk("allow_instance_migration"); ok {
+		data["allow_instance_migration"] = v
+	}
+	if v, ok := d.GetOk("disallow_reauthentication"); ok {
+		data["disallow_reauthentication"] = v
+	}
+
+	log.Printf("[DEBUG] Reading tag data %q from Vault", path)
+	secret, err := client.Logical().Write(path, data)
```

Comment:
`client.Logical().Write()` seems odd here, shouldn't it be `client.Logical().Read()`?  That's what the other two data sources have... ?

Reply:
Because role tags [require a `POST`](https://www.vaultproject.io/api/auth/aws/index.html#create-role-tags) request, `Write()` is needed there. `Read()` would use a `GET` request.